src/BSON/Binary.c

```diff
@@ -142,6 +152,32 @@ static void php_phongo_binary_free_object(void *object TSRMLS_DC) /* {{{ */
 	efree(intern);
 } /* }}} */
 
+zend_object* php_phongo_binary_create_object(zend_class_entry *class_type TSRMLS_DC) /* {{{ */
+{
+        php_phongo_binary_t *intern;
+
+        intern = (php_phongo_binary_t *)ecalloc(1, sizeof(php_phongo_binary_t)+zend_object_properties_size(class_type));
+
+        zend_object_std_init(&intern->std, class_type TSRMLS_CC);
+        object_properties_init(&intern->std, class_type);
+
+        intern->std.handlers = &php_phongo_handler_binary;
+
+        return &intern->std;
+} /* }}} */
+#else
+static void php_phongo_binary_free_object(void *object TSRMLS_DC) /* {{{ */
+{
```

Comment:
Ahhhhh. You are right, sorry.

Maybe add a new one then, one that does the cast (for php5) or calls _fetch_object(zo) for php7.
```
#define Z_OBJ_BINARY(zo) php_binary_fetch_object(zo)
else
#define Z_OBJ_BINARY(zo) (php_phongo_binary_t *)zo
```
should work?